hw/top_earlgrey/data/ip/chip_kmac_testplan.hjson

```diff
@@ -0,0 +1,281 @@
+            '''
+      stage: V2
+      tests: ["chip_sw_lc_ctrl_transition"]
+    }
+    {
+      name: chip_sw_kmac_app_rom
+      desc: '''Verify the ROM interface to KMAC.
+
+            - Backdoor initialize ROM memory immediately out of reset.
+            - ROM will send message to the KMAC containing its memory contents,
+            - KMAC will hash and return the digest to the ROM.
+            - ROM will compare received digest against its first 8 logical memory lines for
+            correctness.
+
+            X-ref'ed with ROM_CTRL test/env.
+            '''
+      stage: V2
+      tests: ["chip_sw_kmac_app_rom"]
+    }
+    {
+      name: chip_sw_kmac_entropy
+      desc: '''Verify the EDN interface to KMAC.
+
+            Requires `EnMasking` parameter to be enabled.
+            SW randomly configures the KMAC in any hashing mode/strength, and enable EDN mode.
+            Randomly enable/disable the `entropy_refresh`.
+            Randomly configure `wait_timer` values (zero for disable, non-zero for timer expire).
+              - Program `wait_timer` to a small value.
+                Check if EDN timeout error occurs after issuing the hashing op.
+              - Adjust `wait_timer` greater than expected EDN latency (with correct `prescaler`
+                too). Then check if Digest is correct.
+            KMAC should send EDN request after `entropy_ready` is set.
+            KMAC also should send out another request to EDN when either:
+              - kmac hash counter hits the configured threshold (assuming it is non-zero)
+              - Hash count exceeds the threshold.
+            SW verifies that KMAC produces the correct digest value.
+
+            TODO: This is pending security review discussion. It is unclear if this feature will be
+            implemented.
+
+            X-ref'ed with EDN test/env.
+            '''
+      stage: V2
+      tests: ["chip_sw_kmac_entropy"]
+    }
+    {
+      name: chip_sw_kmac_idle
+      desc: '''Verify the KMAC idle signaling to clkmgr.
+
+            - Write the KMAC clk hint to 0 within clkmgr to indicate KMAC clk can be gated and
+              verify that the KMAC clk hint status within clkmgr reads 0 (KMAC is disabled).
+            - Write the KMAC clk hint to 1 within clkmgr to indicate KMAC clk can be enabled.
+              Verify that the KMAC clk hint status within clkmgr reads 1 (KMAC is enabled).
+            - Initiate a KMAC operation with a known key, plain text and digest.
+              Write KMAC clock hint to 0 and verify the KMAC clk hint status within clkmgr reads 1
+              (KMAC is enabled), before the KMAC operation is complete.
+            - After the KMAC operation is complete, verify the digest for correctness. Verify that
+              the KMAC clk hint status within clkmgr now reads 0 again (KMAC is disabled).
+            '''
+      stage: V2
+      si_stage: SV3
+      tests: ["chip_sw_kmac_idle"]
+      bazel: []
```

Comment:
Should every test point that as a `si_stage` field, also have `bazel`, `lc_states`, and `features` fields? (i.e. if a test point will be covered in SiVal, it should cover a feature and have (a) bazel target(s)), or

Is it possible for a test point to be pre-Si only?

Reply:
I think in some cases it may make sense for a test point to be pre-si only. For example, in cases in which the test implementation is expected to verify the internal state of the device in a way that is not observable in silicon.

In order to collect metrics for silicon validation, I agree it makes sense to require `bazel`, and `features` to be required whenever `si_stage` is present.

Reply:
should `lc_states` also be required? or are all states assumed if none are specified?

Reply:
side note: eventually, it would be nice to enforce these requirements at the tooling level

Reply:
Added `lc_states` here. I will look into creating a presubmit check to verify all test plan fields.